verde/utils.py

```diff
@@ -207,6 +209,141 @@ def maxabs(*args, nan=True):
     return npmax(absolute)
 
 
+def make_xarray_grid(
+    coordinates,
+    data,
+    data_name,
+    dims=("northing", "easting"),
+    extra_coords_name=None,
+):
+    """
+    Create a :class:`xarray.Dataset` out of gridded data
+
+    The functions creates a :class:`xarray.Dataset` out of 2d gridded data
+    including easting and northing coordinates, any extra coordinates (like
+    upward) and data arrays.
+
+    Parameters
+    ----------
+    coordinates : tuple of arrays
+        Arrays with coordinates of each point in the grid. Each array must
+        contain values for a dimension in the order: easting, northing,
+        vertical, etc. All arrays must be 2d and need to have the same *shape*.
+        These coordinates can be generated through
+        :func:`verde.grid_coordinates`.
+    data : array or tuple of arrays
+        Array or tuple of arrays with data values on each point in the grid.
+        Each array must contain values for a dimension in the same order as
+        the coordinates. All arrays need to have the same *shape*.
+    data_name : str or list
+        String or list containing strings with the names of the data arrays.
```

Comment:
This is inconsistent with `grid` which takes `data_names` as a list always. Should we keep that behaviour to be consistent?

Reply:
Yes, you're right! Firstly I designed it thinking to let it be also a `str`, but then I decided to keep the same behaviour as the `data_name` parameter of the `grid` method, just forgot to change the docs.

Reply:
I'm not entirely sure. I would use this function to create Dataset with a single data array most of the times, so it's a little bit annoying to remember to add the list brackets just to specify the `data_name`. In fact, the parameter name `data_name` doesn't suggest that I should pass several names. So, what if we extend the functionality to accept a single string?